RoslynPluginGenerator/AnalyzerPluginGenerator.cs

```diff
@@ -44,77 +43,88 @@ public class AnalyzerPluginGenerator
+            }
             if (logger == null)
             {
                 throw new ArgumentNullException("logger");
             }
+            this.packageHandler = packageHandler;
             this.logger = logger;
         }
 
-        public bool Generate(NuGetReference analyzeRef, string sqaleFilePath)
+        public bool Generate(NuGetReference analyzeRef, string language, string sqaleFilePath)
         {
             // sqale file path is optional
             if (analyzeRef == null)
             {
                 throw new ArgumentNullException("analyzeRef");
             }
+            if (string.IsNullOrWhiteSpace(language))
+            {
+                throw new ArgumentNullException("language");
+            }
+            SupportedLanguages.ThrowIfNotSupported(language);
 
-            string baseDirectory = Path.Combine(
-                Path.GetTempPath(),
-                Assembly.GetEntryAssembly().GetName().Name);
-
-            string nuGetDirectory = Path.Combine(baseDirectory, ".nuget");
-
-            NuGetPackageHandler downloader = new NuGetPackageHandler(logger);
-
-            IPackage package = downloader.FetchPackage(NuGetPackageSource, analyzeRef.PackageId, analyzeRef.Version, nuGetDirectory);
+            string nuGetDirectory = Utilities.CreateTempDirectory(".nuget");
+            IPackage package = this.packageHandler.FetchPackage(analyzeRef.PackageId, analyzeRef.Version, nuGetDirectory);
 
             if (package != null)
             {
-                // TODO: support multiple languages
-                string language = SupportedLanguages.CSharp;
-                PluginManifest pluginDefn = CreatePluginDefinition(package);
+                // Create a uniquely-named temp directory for this generation run
+                string baseDirectory = Utilities.CreateTempDirectory(".gen");
+                baseDirectory = CreateSubDirectory(baseDirectory, Guid.NewGuid().ToString());
+                this.logger.LogDebug(UIResources.APG_CreatedTempWorkingDir, baseDirectory);
 
-                string outputDirectory = Path.Combine(baseDirectory, ".output", Guid.NewGuid().ToString());
-                Directory.CreateDirectory(outputDirectory);
-
-                string rulesFilePath = Path.Combine(outputDirectory, "rules.xml");
+                PluginManifest pluginDefn = CreatePluginDefinition(package);
 
-                // TODO: we shouldn't try to work out where the content files have been installed by NuGet.
-                // Instead, we should use the methods on IPackage to locate the assemblies e.g. Package.GetFiles()
-                string packageDirectory = Path.Combine(nuGetDirectory, package.Id + "." + package.Version.ToString());
-                Debug.Assert(Directory.Exists(packageDirectory), "Expected package directory does not exist: {0}", packageDirectory);
+                string rulesFilePath = Path.Combine(baseDirectory, "rules.xml");
```

Comment:
You were previously naming to the rules file as ID + rules.xml, but I assume you'll move / rename this file